src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -412,6 +415,32 @@ export class Evaluator {
     this.nextPositionsToUpdate.addMany(invalidatedPositions);
   }
 
+  private invalidatePendingPositionsStack() {
+    if (this.positionsToInvalidate.length > 0) {
+      // the result matrices are split in subzones to exclude the array formula position.
+      // This avoids to invalidate the topleft position of an array formula
+      // and create an infinite loop of self-invalidation
+      const boundingboxes = this.positionsToInvalidate
+        .map((range) =>
+          excludeTopLeft(range.zone).map((zone) => ({
+            sheetId: range.sheetId,
+            zone,
+          }))
+        )
+        .flat();
+      const invalidatedPositions = this.formulaDependencies().getCellsDependingOn(boundingboxes);
+      for (const range of this.positionsToInvalidate) {
+        invalidatedPositions.delete({
```

Comment:
this is causing the issue @hokolomopo posted on the task chatter.
A1 should be recomputed (because A3 spilled on the zone it depends on), but this line removes it from `this.nextPositionsToUpdate`